compiler/tests/fixtures/demos/uniswap.fe

```diff
@@ -0,0 +1,359 @@
+            # `*` never overflows, and + overflow is desired
+            # todo: the library `UQ112x112` is used here in the sol implementation
+            self.price0CumulativeLast += (reserve1 / reserve0) * time_elapsed
+            self.price1CumulativeLast += (reserve0 / reserve1) * time_elapsed
+
+        reserve0 = u112(self.balance0)
+        reserve1 = u112(self.balance1)
+        self.block_timestamp_last = block_timestamp
+        emit Sync(reserve0, reserve1)
+
+    # if fee is on, mint liquidity equivalent to 1/6th of the growth in sqrt(k)
+    def _mint_fee(reserve0: u112, reserve1: u112) -> bool:
+        fee_to: address = UniswapV2Factory.load(factory).fee_to()
+        fee_on: bool = fee_to != address(0)
+        k_last: u256 = self.k_last # gas savings
+        if fee_on:
+            if k_last != 0:
+                root_k: u256 = Math.sqrt(reserve0 * reserve1)
+                root_k_last: u256 = Math.sqrt(k_last)
+                if root_k > root_k_last:
+                    numerator: u256 = total_supply * root_k - root_k_last
+                    denominator: u256 = root_k * 5 + root_k_last
+                    liquidity: u256 = numerator / denominator
+                    if liquidity > 0:
+                        self._mint(fee_to, liquidity)
+        elif k_last != 0:
+            self.k_last = 0
+
+        return fee_on
+
+    # this low-level function should be called from a contract which performs important safety checks
+    def mint(to: address) -> u256:
+        # (reserve0: u112, reserve1: u112) = self.get_reserves() # gas savings
+        balance0: u256 = ERC20.load(token0).balanceOf(self.address)
+        balance1: u256 = ERC20.load(token1).balanceOf(self.address)
+        amount0: u256 = balance0.sub(reserve0)
+        amount1: u256 = balance1.sub(reserve1)
+
+        fee_on: bool = self._mint_fee(reserve0, reserve1)
+        total_supply: u256 = self.total_supply # gas savings, must be defined here since totalSupply can update in _mintFee
+        liquidity: u256 = 0
+        if total_supply == 0:
+            liquidity = Math.sqrt(amount0 * amount1) - MINIMUM_LIQUIDITY
+            self._mint(address(0), MINIMUM_LIQUIDITY) # permanently lock the first MINIMUM_LIQUIDITY tokens
+        else:
+            liquidity = Math.min(amount0.mul(_totalSupply) / _reserve0, amount1.mul(_totalSupply) / _reserve1)
+
+        assert liquidity > 0, "UniswapV2: INSUFFICIENT_LIQUIDITY_MINTED"
+
+        self._mint(to, liquidity)
+        self._update(balance0, balance1, reserve0, reserve1)
+
+        if fee_on:
+            self.k_last = reserve0 * reserve1 # reserve0 and reserve1 are up-to-date
+
+        emit Mint(msg.sender, amount0, amount1)
+
+    # this low-level function should be called from a contract which performs important safety checks
+    pub def burn(to: address): # -> (u256, u256):
+        # (reserve0: u112, reserve1: u112) = self.get_reserves() # gas savings
+        token0: ERC20 = ERC20.load(self.token0)
```

Comment:
`load` is the extra keyword needed to typecast explicitly for the contracts?

Reply:
I was thinking in terms of rust's `From` trait here, but used the term `load` instead of `from`. Personally, I think having typecasts of the form `Type(expression)` scattered around makes code somewhat difficult to read.

@cburgdorf, what do you think? You've done most of the work on typecasting so far and usually have good thoughts on syntax in general.

Reply:
I don't have a strong opinion here but I would mildly prefer to go with `ERC20(<address>)` instead of the `ERC20.load(<address>)` syntax.

Reply:
I'll change it to `ERC20(address)` for now and we can see how it works.